validation/pytorch.py

```diff
@@ -40,16 +41,21 @@
 
 def self_consistency(rnn, x):
   x_cuda = x.clone().cuda()
-  x_cpu = x.clone().cpu()
+  x_cuda_torch = x_cuda.detach().clone()
```

Comment:
I'm curious why you chose to mock `_is_cuda` on the RNN instead of passing in a CPU tensor directly. Was there an issue with the previous approach?

Reply:
[Currently *PyTorch* cannot get identical random masks](https://pytorch.org/docs/1.10.1/notes/randomness.html) (*zoneout* masks or *dropout* masks) on `"cpu"` and `"cuda"`. So we compare _PyTorch_ version *RNN*s on `"cuda"` with haste *RNN*s, which only have GPU implementation.

To run the _PyTorch_ version *RNN*s on `"cuda"`, maybe we could create RNNs like:

```python
pytorch_rnn = haste.GRU(32, 256, backend="pytorch").cuda()
haste_rnn = haste.GRU(32, 256, backend="haste").cuda()
```

But the changes of interfaces seems to be off topic. It seems to me that mocking `_is_cuda` is the simplest way to run *PyTorch* version `RNN`s on `"cuda"`.

Reply:
I see, thanks for clarifying. Your approach looks right.